aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -705,6 +705,73 @@ struct ReductionMulOp {
   __forceinline__ scalar_t identity_cpu() const { return 1; }
 };
 
+void _apply_sparse_csr_linear_solve(
+  const Tensor& A,
+  const Tensor& b,
+  const bool left,
+  const Tensor& x) {
+#if defined(USE_ROCM) || !defined(USE_CUDSS)
+  TORCH_CHECK(
+      false,
+      "Calling linear solver with sparse tensors requires compiling ",
+      "PyTorch with CUDA cuDSS and is not supported in ROCm build.");
+#else
+  // layout check
+  TORCH_CHECK(A.is_sparse_csr(), "A must be a CSR matrix");
+  TORCH_CHECK(b.layout() == kStrided, "b must be a strided tensor");
+  TORCH_CHECK(x.layout() == kStrided, "x must be a strided tensor");
+  // dim check
+  TORCH_CHECK(b.dim() == 1, "b must be a 1D tensor");
+  TORCH_CHECK(b.stride(0) == 1, "b must be a column major tensor");
+  TORCH_CHECK(b.size(0) == A.size(0), "linear system size mismatch.");
+  TORCH_CHECK(x.dim() == 1, "x must be a 1D tensor");
+  TORCH_CHECK(x.stride(0) == 1, "x must be a column major tensor");
+  TORCH_CHECK(x.size(0) == A.size(1), "linear system size mismatch.");
+  TORCH_CHECK(A.dtype() == b.dtype() && A.dtype() == x.dtype(), "A, x, and b must have the same dtype");
+  TORCH_CHECK(left == true, "only left == true is supported by the Sparse CSR backend")
+
+  Tensor crow = A.crow_indices();
+  Tensor col = A.col_indices();
+  if (crow.scalar_type() != ScalarType::Int) {
+    crow = crow.to(crow.options().dtype(ScalarType::Int));
+    col = col.to(col.options().dtype(ScalarType::Int));
+  }
+  int* rowOffsets = crow.data<int>();
+  int* colIndices = col.data<int>();
+  Tensor values = A.values();
+  // cuDSS data structures and handle initialization
+  cudssConfig_t config;
+  cudssMatrix_t b_mt;
+  cudssMatrix_t A_mt;
+  cudssMatrix_t x_mt;
+  cudssData_t cudss_data;
+  cudssHandle_t handle = at::cuda::getCurrentCudssHandle();
+
+  TORCH_CUDSS_CHECK(cudssConfigCreate(&config));
+  TORCH_CUDSS_CHECK(cudssDataCreate(handle, &cudss_data));
+
+  AT_DISPATCH_FLOATING_TYPES(values.type(), "create_matrix", ([&] {
+    scalar_t* values_ptr = values.data<scalar_t>();
+    scalar_t* b_ptr = b.data<scalar_t>();
+    scalar_t* x_ptr = x.data<scalar_t>();
+    auto CUDA_R_TYP = std::is_same<scalar_t, double>::value ? CUDA_R_64F : CUDA_R_32F;
+    TORCH_CUDSS_CHECK(cudssMatrixCreateDn(&b_mt, b.size(0), 1, b.size(0), b_ptr, CUDA_R_TYP, CUDSS_LAYOUT_COL_MAJOR));
+    TORCH_CUDSS_CHECK(cudssMatrixCreateDn(&x_mt, x.size(0), 1, x.size(0), x_ptr, CUDA_R_TYP, CUDSS_LAYOUT_COL_MAJOR));
+    TORCH_CUDSS_CHECK(cudssMatrixCreateCsr(&A_mt, A.size(0), A.size(1),  A._nnz(), rowOffsets, rowOffsets + crow.size(0), colIndices, values_ptr, CUDA_R_32I, CUDA_R_TYP, CUDSS_MTYPE_GENERAL, CUDSS_MVIEW_FULL, CUDSS_BASE_ZERO));
+  }));
+  cudssAlgType_t reorder_alg = CUDSS_ALG_1;
+  cudssConfigSet(config, CUDSS_CONFIG_REORDERING_ALG, &reorder_alg, sizeof(cudssAlgType_t));
```

Comment:
Hi! Could you comment on the necessity of this? Are there particular test cases which cannot pass with the the default algorithm (same as setting `CUDSS_ALG_DEFAULT` or not setting anything at all)
The default reordering algorithm (METIS-based) has better perf in most of the cases, sometimes significantly (but admittedly can have much more perturbed pivots and less accuracy sometimes).

Also, if trying the default algorithm, I suggest making 2 steps of iterative refinement by default (which can help with perturbed pivots for `CUDSS_ALG_DEFAULT` reordering algorithm) via `cudssConfigSet()` with `CUDSS_CONFIG_IR_N_STEPS`.
For `CUDSS_ALG_1` reordering, iterative refinement likely is not needed.

Reply:
When I was using the cudss 0.2 version, the `cudssExecute(handle, CUDSS_PHASE_ANALYSIS, config, cudss_data, A_mt, x_mt, b_mt)` statement using `CUDSS_ALG_DEFAULT` always return a status of internal error and fail the `TORCH_CUDSS_CHECK`, even though the solution it gives is correct. 
Looks like the bug has been fixed in 0.3 so I am going to switch back to `CUDSS_ALG_DEFAULT`.
Thanks for the input!